Gems/Prefab/PrefabBuilder/PrefabGroup/DefaultProceduralPrefab.cpp

```diff
@@ -0,0 +1,553 @@
+            const auto thisNodeIndex = nodeEntity.first;
+            auto parentNodeIndex = graph.GetNodeParent(thisNodeIndex);
+            while (parentNodeIndex.IsValid())
+            {
+                auto parentNodeIterator = nodeDataMap.find(parentNodeIndex);
+                if (nodeDataMap.end() != parentNodeIterator)
+                {
+                    auto parentEntiyIterator = nodeEntityMap.find(parentNodeIterator->first);
+                    if (nodeEntityMap.end() != parentEntiyIterator)
+                    {
+                        parentEntityId = parentEntiyIterator->second;
+                        break;
+                    }
+                }
+                else if (graph.HasNodeParent(parentNodeIndex))
+                {
+                    parentNodeIndex = graph.GetNodeParent(parentNodeIndex);
+                }
+                else
+                {
+                    parentNodeIndex = {};
+                }
+            }
+
+            AZ::Entity* entity = AZ::Interface<AZ::ComponentApplicationRequests>::Get()->FindEntity(nodeEntity.second);
+            auto* entityTransform = entity->FindComponent<AzToolsFramework::Components::TransformComponent>();
+            if (!entityTransform)
+            {
+                return {};
+            }
+
+            // parent entities
+            if (parentEntityId.IsValid())
+            {
+                entityTransform->SetParent(parentEntityId);
+            }
+
+            auto thisNodeIterator = nodeDataMap.find(thisNodeIndex);
+            AZ_Assert(thisNodeIterator != nodeDataMap.end(), "This node index missing.");
+            auto thisTransformIndex = thisNodeIterator->second.m_transformIndex;
+
+            // get node matrix data to set the entity's local transform
+            const auto nodeTransform = azrtti_cast<const DataTypes::ITransform*>(graph.GetNodeContent(thisTransformIndex));
+            if (nodeTransform)
+            {
+                entityTransform->SetLocalTM(AZ::Transform::CreateFromMatrix3x4(nodeTransform->GetMatrix()));
+            }
+            else
+            {
+                entityTransform->SetLocalTM(AZ::Transform::CreateUniformScale(1.0f));
+            }
+
+            PrefabGroupNotificationBus::Broadcast(&PrefabGroupNotificationBus::Events::OnUpdatePrefabEntity, nodeEntity.second);
+        }
+
+        return entities;
+    }
+
+    bool DefaultProceduralPrefabGroup::CreatePrefabGroupManifestUpdates(
+        ManifestUpdates& manifestUpdates,
+        const Containers::Scene& scene,
+        const EntityIdList& entities,
+        const AZStd::string& filenameOnly,
+        const AZStd::string& relativeSourcePath) const
+    {
+        AZStd::string prefabTemplateName{ relativeSourcePath };
+        AZ::StringFunc::Path::ReplaceFullName(prefabTemplateName, filenameOnly.c_str());
+        AZ::StringFunc::Replace(prefabTemplateName, "\\", "/"); // the source folder uses forward slash
```

Comment:
is this true on all platforms?

Reply:
yep,  this is an asset path which are always forward slashes